commands/run.sh

```diff
@@ -95,6 +96,25 @@ while IFS=$'\n' read -r vol ; do
   [[ -n "${vol:-}" ]] && run_params+=("-v" "$(expand_relative_volume_path "$vol")")
 done <<< "$(plugin_read_list VOLUMES)"
 
+# Mount ssh-agent socket and known_hosts
+if [[ "${BUILDKITE_PLUGIN_DOCKER_COMPOSE_MOUNT_SSH_AGENT:-$mount_ssh_agent}" =~ ^(true|on|1)$ ]] ; then
+  if [[ -z "${SSH_AUTH_SOCK:-}" ]] ; then
+    echo "+++ 🚨 \$SSH_AUTH_SOCK isn't set, has ssh-agent started?"
+    exit 1
+  fi
+  if [[ -z "${IS_IN_TEST:-}" ]]; then
+    if [[ ! -S "${SSH_AUTH_SOCK}" ]] ; then
+      echo "+++ 🚨 The file at ${SSH_AUTH_SOCK} isn't a socket, has ssh-agent started?"
+      exit 1
+    fi
+  fi
+  run_params+=(
+    "-e" "SSH_AUTH_SOCK=/ssh-agent"
+    "--volume" "${SSH_AUTH_SOCK}:/ssh-agent"
+    "--volume" "${HOME}/.ssh/known_hosts:/root/.ssh/known_hosts"
```

Comment:
I guess this should take https://github.com/buildkite-plugins/docker-compose-buildkite-plugin#user-optional-run-only into account (do not assume the user is always `root`).

Reply:
> I guess this should take https://github.com/buildkite-plugins/docker-compose-buildkite-plugin#user-optional-run-only into account (do not assume the user is always `root`).

your welcome to make the change. I'm no longer at the company that uses buildkite, so I'm not going to spend time on the fix sorry.